packages/react-input/Spec.md

```diff
@@ -1,63 +1,322 @@
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+See [Structure section](#structure) for the output HTML.
+
+```tsx
+<Label htmlFor="input1">Label</Label> // note that this is not built in
+<Input
+  className="rootClass"
+  input={{ className: 'inputClass' }}
+  id="input1"
+  value="something"
+  onChange={(ev, data) => console.log(data.value)}
+  bookendBefore="bookendBefore"
+  bookendAfter="bookendAfter"
+  insideBefore={<SearchIcon />}
+  insideAfter={<ClearIcon />}
+/>
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+### Visual variants
+
+See the design spec for details on the appearance of each of these. Note that the actual colors used will follow the theme; white/gray/black are just examples from the default theme.
+
+Visual variants for the main field are as follows:
+
+- `outline` (default): the field has a full outline, with a slightly darker underline on the bottom
+- `underline`: the field has an underline on the bottom only
+- `filledDarker`: the field has a gray background and no underline/outline
+- `filledLighter`: the field has a white background and no underline/outline (for use on top of gray/colored backgrounds)
+
+Visual variants for the bookend are as follows:
+
+- `filled` (default): bookends have a gray background, black text, and no border
+- `brand`: bookends have a brand color (such as blue or purple) background, white text, and no border
+- `transparent`: bookends have a transparent background, black text, and only a thin inside border to distinguish them from the field
+
+### Behavior variants
+
+In the future we may implement behavior variants, such as a password field with reveal button, but that will be spec'd if/when we need it. See [Spec-variants.md](./Spec-variants.md) for more details.
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+In this component, `input` is the primary slot. See notes under [Structure](#structure).
+
+```ts
+export type InputProps = InputCommons & Omit<ComponentProps<InputSlots, 'input'>, 'children'>;
+```
+
+### Main props
+
+Note: most of these are defined in `InputCommons` since they're shared with `InputState`.
+
+```ts
+export type InputCommons = FieldSizeProps & {
+  /**
+   * If true, the field will have inline display, allowing it be used within text content.
+   * If false (the default), the field will have block display.
+   */
+  inline?: boolean;
+
+  /**
+   * Controls the colors and borders of the field.
+   * @default 'outline'
+   */
+  appearance?: 'outline' | 'underline' | 'filledDarker' | 'filledLighter';
+
+  /**
+   * Controls the colors and borders of the bookends.
+   * @default 'filled'
+   */
+  bookendAppearance?: 'filled' | 'brand' | 'transparent';
+};
```

Comment:
I think we are leaning on doing everything in `[Component]Commons` types required unless there is a specific reason not to and extending `Partial<[Component]Commons>` in `[Component]Props` so they are not required there but we always have to provide a default for them when construction `[Component]State`.

Reply:
That's actually what I'm doing in the implementation right now, I just wasn't sure if it was helpful to go into that detail in the spec (since the state isn't really part of the spec)